File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    static int[] numbers;
    static int answer, target;
    static void dfs(int idx, int num) {
        if(idx == numbers.length) {
            if(num == target) answer++;
            return;
        }
        
        dfs(idx+1, num - numbers[idx]);
        dfs(idx+1, num + numbers[idx]);
    }
    
    public int solution(int[] numbers, int target) {
        this.numbers = numbers;
        this.target = target;
        dfs(0,0);
        
        return answer;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;
int answer = 0;
vector<int> nums;
int t;

void dfs(int idx, int sum) {
    if(idx == (int)nums.size()) {
        if(sum == t) answer++;
        return;
    }
    
    // 더한다
    dfs(idx+1, sum + (-1)*nums[idx]);
    // 뺀다
    dfs(idx+1, sum + nums[idx]);
}

int solution(vector<int> numbers, int target) {
    nums = numbers;
    t = target;
    
    dfs(0, 0);
    
    return answer;
}